assets/js/base/context/utils.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * This is an assertion utility for validating that the incoming value prop
+ * value on a given context provider is valid and throws an error if it isn't.
+ *
+ * Note: this asserts values that are expected to be an object.
+ *
+ * The validationMap is expected to be an object in the following shape.
+ *
+ * {
+ *   [expectedPropertyName<String>]: {
+ *     required: [expectedRequired<Boolean>]
+ *     type: [expectedtype<String>]
```

Comment:
Should we capitalize `type` in `expectedtype` as in `expectedRequired`?
```suggestion
 *     type: [expectedType<String>]
```